pyproject.toml

```diff
@@ -12,18 +12,36 @@ dependencies = [
     "boto3",
     "pandas",
     "levenshtein",
-    "pathlib==1.0.1",
-    "python-dotenv==1.0.1",
-    "scikit-learn==1.4.0",
-    "click==8.1.7",
-    "PyYAML==6.0.1",
-    "langdetect==1.0.9",
+    "pathlib",
+    "python-dotenv",
+    "scikit-learn>=1.4.0",
+    "click>=8.0.0",
+    "PyYAML>=6.0.1",
+    "langdetect>=1.0.9",
+    "regex",
     "backoff",
-    "PyMuPDF==1.23.26",
+    "PyMuPDF>=1.23.26",
     "opencv-python-headless",
-    "quads==1.1.0",
+    "quads>=1.1.0",
 ]
 
+[project.optional-dependencies]
+test = [
+    "pytest==8.1.1",
+    "pytest-cov==5.0.0"
+]
+lint = [
+    "pre-commit==3.6.2",
+]
+experiment_tracking = [
```

Comment:
I had to change to this `experiment-tracking` (with hyphen instead of underscore) to make it work. (It seems that Python automatically normalizes these names to use hyphens instead of underscores.)